src/api.jl

```diff
@@ -349,6 +353,99 @@ end
 # todo: when we have async transactions, add a variation that dispatches and
 #   waits .. consider creating two entry points for readonly and readwrite.
 
+function exec_v2(ctx::Context, database::AbstractString, engine::AbstractString, source; inputs = nothing, readonly = false, kw...)
```

Comment:
change to `RAI.V2.exec`?

Reply:
👍 but actually: i think this package is so early days, i'd rather just call this one _exec_, and rename the v1 exec to something else, maybe `RAI.V1.exec` or `RAI.exec_v1`!

(Or actually, naming this one `exec_async()` and then having one more wrapper called `exec()` that does the polling)

Reply:
For now i'm going to rename this to `exec_async()`, based on the above, i think.